exec/activity-connection-saver.go

```diff
@@ -24,9 +24,17 @@ const (
 // To prevent close exec connection
 // (https://blog.openshift.com/executing-commands-in-pods-using-k8s-api/ - Connection lifecycle)
 // let's send empty byte array each 30 sec.
-func saveActivity(machineExec *model.MachineExec) {
+func saveActivity(machineExec *model.MachineExec, stopActivitySaver chan bool) {
 	ticker := time.NewTicker(ActivityTimeOut * time.Second)
-	for range ticker.C {
-		machineExec.MsgChan <- make([]byte, 0)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-stopActivitySaver:
+			return
+		case <-ticker.C:
+			machineExec.MsgChan <- make([]byte, 0)
+			break
```

Comment:
Just interesting if I miss something: Does this break do anything? Just explicitly shows - process to next iteration?

Reply:
Well, Good catch, this `break` do nothing in this case.  I will remove it.
In general `break` inside`select` could terminate some some part of the case logic:

```golang
for i:= 0; i < 5; i++ {
      switch i {
	  case 1, 2:
		  fmt.Println("Test")
		  if i == 2 {
			  break
		  }
		  fmt.Println("Go on")
	  default:
		 fmt.Println(i)
      }
  }
---------
Output:
0
Test
Go on
Test
3
4
```
Also you  could terminate external `for` using label:
```golang
L:
 for i:= 0; i < 100; i++ {
     switch i {
     case 5:
            break L
    default:
            fmt.Println(i)
     }
 }
---------
Output:
0
1
2
3
4
```

Reply:
I think you know all this info.